### PR TITLE
Make sure esbuild warnings are not hidden on deploy

### LINKED ISSUE #4923: 🐛 BUG: Insufficient warning reporting in `wrangler deploy`
### Which Cloudflare product(s) does this pertain to?

Workers Runtime, Wrangler core

### What version(s) of the tool(s) are you using?

3.26.0

### What version of Node are you using?

20.10.0

### What operating system and version are you using?

Windows 11

### Observed behavior
When running the command `wrangler dev` a warning is produced if a semicolon is automatically inserted.

When running the command `wrangler deploy`, no warnings are produced if a semicolon is automatically inserted.

### Expected behavior
When running the command `wrangler deploy`, warnings should be produced if potentially breaking changes are made to the source code (such as but not limited to semicolon insertion).

### Steps to reproduce
Steps provided are for Windows. I'm sure they are not difficult to adapt to a Unix based system.
- Download [SampleFiles.zip](https://github.com/cloudflare/workers-sdk/files/14167272/SampleFiles.zip)
- Open command prompt and `cd` to the downloaded folder
- Run command `npm i`
- Run command `npx wrangler dev ./Program.js -l`
   - Observe a warning is logged regarding `semicolon-after-return`, a breaking change for this code.
- Run command `npx wrangler deploy` (authenticate if necessary)
   - No warnings are logged, but a breaking change was introduced.

Note the whitespace around lines 27-29 of MyHelper.js. This is a simplified version of code I wrote and spent quite a bit of time debugging why the method returned `undefined` when the source code only returned a `Boolean`. The whitespace was to keep line length reasonable while also somewhat aligning the two similar comparisons. I intend to introduce tooling to our CI/CD such that warnings are treated as errors during deployment, but I can only do that if warnings are logged in the first place.

This is especially concerning because one can have automated tests verifying the source code, but if a breaking change is silently introduced during deployment, we might not notice before a customer is impacted.

### Please provide a link to a minimal reproduction

https://github.com/cloudflare/workers-sdk/files/14167272/SampleFiles.zip

### Please provide any relevant error logs

Actual deploy logs:
```
Total Upload: 2.20 KiB / gzip: 0.85 KiB
Uploaded semicolon-sample (0.81 sec)
Published semicolon-sample (3.49 sec)
  https://semicolon-sample.REDACTED.workers.dev
Current Deployment ID: 3cd4304e-22d5-4efe-a2da-6915f97afec8
```

Desired deploy logs:
```
▲ [WARNING] The following expression is not returned because of an automatically-inserted semicolon [semicolon-after-return]

    MyHelper.js:27:26:
      27 │                     return
         ╵                           ^
Total Upload: 2.20 KiB / gzip: 0.85 KiB
Uploaded semicolon-sample (0.81 sec)
Published semicolon-sample (3.49 sec)
  https://semicolon-sample.REDACTED.workers.dev
Current Deployment ID: 3cd4304e-22d5-4efe-a2da-6915f97afec8
```

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Make sure esbuild warnings are logged when running `wrangler deploy`. 

Fixes #4923

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required / Maybe required
  - [x] Not required because: this behaviour has no e2e tests
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Not necessary because: bugfix

### CODE REVIEW

File: packages/wrangler/src/deploy/deploy.ts [hunk 1]
```diff
@@ -570,19 +572,6 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
...
-				module.filePath === undefined
-					? module.name
-					: path.relative("", module.filePath);
-			const bytesInOutput =
-				typeof module.content === "string"
-					? Buffer.byteLength(module.content)
-					: module.content.byteLength;
-			dependencies[modulePath] = { bytesInOutput };
-		}
-
```
Comment (RamIdeas) [inline packages/wrangler/src/deploy/deploy.ts:585]: Is this deletion intentional?

Comment (penalosa) [reply packages/wrangler/src/deploy/deploy.ts:585]: Yes, this block of code was duplicated—probably from a bad rebase/merge?

File: packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+export function logBuildOutput(
+	nodejsCompatMode: NodeJSCompatMode | undefined,
+	onStart?: () => void,
+	updateBundle?: () => void
+): Plugin {
+	let bundled = false;
+
+	return {
+		name: "log-build-output",
+		setup(b) {
```
Comment (andyjessop) [inline packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts:18]: Nit: Is `b` `build`?

File: packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+			b.onStart(() => {
+				onStart?.();
+			});
+			b.onEnd(async (result) => {
+				const errors = result.errors;
```
Comment (andyjessop) [inline packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts:23]: Nit: Destructure the function arg instead perhaps?

**Review by andyjessop — APPROVED** [review]

**Review by RamIdeas — APPROVED** [review]

File: packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts (hunk 3 above)
Comment (penalosa) [reply packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts]: https://github.com/cloudflare/workers-sdk/pull/6566/commits/0d6eeb9fa8f6831cad98d3934ff38faf0c5a3b0a

File: packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts (hunk 2 above)
Comment (penalosa) [reply packages/wrangler/src/deployment-bundle/esbuild-plugins/log-build-output.ts]: https://github.com/cloudflare/workers-sdk/pull/6566/commits/0d6eeb9fa8f6831cad98d3934ff38faf0c5a3b0a

**Pull request merged**